source/Government.cpp

```diff
@@ -121,8 +123,11 @@ void Government::Load(const DataNode &node)
 		{
 			if(key == "provoked on scan")
 				provokedOnScan = false;
-			else if(key == "raid")
-				raidFleet = nullptr;
+			else if(key == "raid" && !hasValue)
+			{
+				raidFleets.clear();
+				notClearedRaids = false;
+			}
```

Comment:
The bottom of this if/else tree has a print trace saying you can't remove from the given value. We enter this tree if the line starts with `remove`, but we're skipping over this raid clear if we're given a specific raid to remove. That means if we have `remove raid "Pirate Raid"`, it'll enter `if(remove)`, skip over this `key == "raid"` because `!hasValue` is false, then print a trace saying you can't remove the given value, even if you can. Then, because the `if(remove)` is attached to all the other key checks by `else if`s, we never hit the normal `key == "raid"` code.

This is all set up this way because Government hasn't had a key that you can remove a single value of before. It's always either been overwrite what was there or completely clear it out. Look to a class like System for a place where `remove` behavior is different depending on if a value is given or not.

Reply:
Cant remove in general have !hasValue added to it?

Reply:
That should work, yes.

Reply:
Actually that would make the error parsing less precise than it used to be I think.
So I made it the same way as it works in system +- (no need for a whole vector here)